rpm/SPECS/docker-ce.spec

```diff
@@ -94,15 +85,51 @@ fi
 if ! getent group docker > /dev/null; then
     groupadd --system docker
 fi
+dbefile=/var/lib/docker/distribution_based_engine.json
+URL=https://docs.docker.com/releasenote
+if [ -f "${dbefile}" ] && sed -e 's/.*"platform"[ \t]*:[ \t]*"\([^"]*\)".*/\1/g' "${dbefile}"| grep -v -i community > /dev/null; then
+    echo
+    echo
+    echo
+    echo "Warning: Your engine has been activated to Docker Engine - Enterprise but you are still using Community packages"
+    echo "You can use the 'docker engine update' command to update your system, or switch to using the Enterprise packages."
+    echo "See $URL for more details."
+    echo
+    echo
+    echo
+else
+    rm -f %{_bindir}/dockerd
+    update-alternatives --install %{_bindir}/dockerd dockerd %{_bindir}/dockerd-ce 1 \
+        --slave "${dbefile}" distribution_based_engine.json /var/lib/docker/distribution_based_engine-ce.json
```

Comment:
I have a feeling we'll need to prefix the `/var/lib/docker` with some `%{_xxx}` thingie to get alternative install root installs to work properly, but lets punt on that for the moment.